iterators_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,59 @@
 #include "impl.h"
 
+#include <sstream>
+#include <iterator>
+#include <algorithm>
+#include <list>
+
+
+std::string reverseWords(const std::string& input)
+{
+    if (input.empty())
+    {
+        return {};
+    }
+
+    std::istringstream inputStream {input};
+    std::ostringstream outputStream;
+    std::list<std::string> container { std::istream_iterator<std::string>(inputStream),
+                                       std::istream_iterator<std::string>() };
+
+    if (container.empty())
+    {
+        return {};
+    }
+
+    auto rbegin = container.rbegin();
+    auto rend = container.rend();
+    std::copy(rbegin, std::prev(rend), std::ostream_iterator<std::string>(outputStream, " "));
+    std::copy(std::prev(rend), rend, std::ostream_iterator<std::string>(outputStream));
```

Comment:
@smay1613
Would be better just to pop the last space from the result?

Reply:
operator <<